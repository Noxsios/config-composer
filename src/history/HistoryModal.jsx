import { useState, useContext, useEffect } from "react";
import MetaContext from "../MetaContext";
import { EuiConfirmModal, EuiOverlayMask } from "@elastic/eui";
import { useIndexedDB } from "react-indexed-db";

const HistoryModal = () => {
  const { getByID, getAll, clear } = useIndexedDB("history");
  const [isModalVisible, setIsModalVisible] = useState(false);
  const { meta, setMeta } = useContext(MetaContext);

  useEffect(() => {
    getAll().then((entries) => {
      if (
        entries.length !== 0 &&
        meta.json === "undefined" &&
        meta.markdown === "undefined"
      ) {
        setIsModalVisible(true);
      } else {
        setIsModalVisible(false);
      }
    });
  }, [getAll, meta, isModalVisible]);

  const closeModal = () => {
    clear().then(() => {
      setIsModalVisible(false);
    });
  };

  const closeAndUseLast = () => {
    getByID(0).then((entry) => {
      setMeta({ ...entry });
      setIsModalVisible(false);
    });
  };

  return (
    <div className="no-print">
      {isModalVisible && (
        <EuiOverlayMask>
          <EuiConfirmModal
            maxWidth={500}
            title="Pick up where you left off?"
            onCancel={closeModal}
            confirmButtonText="Yes"
            cancelButtonText="No"
            onConfirm={closeAndUseLast}
          />
        </EuiOverlayMask>
      )}
    </div>
  );
};

export default HistoryModal;
