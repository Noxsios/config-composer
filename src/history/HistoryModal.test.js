import renderer from "react-test-renderer";
import HistoryModal from "./HistoryModal";

it("renders history modal skeleton", () => {
  const t = renderer.create(<HistoryModal />).toJSON();
  expect(t).toMatchSnapshot();
});
