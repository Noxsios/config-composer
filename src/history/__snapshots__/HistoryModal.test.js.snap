// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders history modal skeleton 1`] = `
<div
  className="no-print"
/>
`;
